### PR TITLE
Freki change to 1.1.0

### PR DESCRIPTION
**Pull request by MohammadBilalArif**
@glaslos  Issue~#48 Fixed.

### CODE REVIEW

**Comment by glaslos** [comment]
We could even register separate handlers per protocol instead of having one for Glutton. Means also we have more rules. Any opinions?

**Comment by MohammadBilalArif** [comment]
Great idea :) But Freki will open a tcp socket per handler, It is good idea to have multiple sockets than one?

**Comment by MohammadBilalArif** [comment]
@glaslos  I thought Freki will open separate socket per protocol but I doesn't work like this :). Well I made some changes in architecture, now we have separate handler per protocol :)

File: glutton.go [hunk 1]
```diff
@@ -5,22 +5,31 @@ import (
 	"net"
 	"os"
 	"path/filepath"
-	"strings"
+	// "strings"
```
Comment (glaslos) [inline glutton.go]: huh? :)

File: glutton.go [hunk 2]
```diff
@@ -51,74 +60,60 @@ func (g *Glutton) makeID() error {
...
-		HTTPAddr: logHTTP,
+	g.addServers()
+	g.mapProtocolHandler()
+
+	g = &Glutton{
+		processor:        processor,
+		logger:           log,
+		rules:            rule,
+		protocolHandlers: make(map[string]protocolHandlerFunc, 0),
+		address:          producer.NewAddress(log, logHTTP),
```
Comment (glaslos) [inline glutton.go]: I don't thin `address` is the correct name for this field anymore.

File: producer/http.go [hunk 3]
```diff
@@ -12,8 +13,8 @@ import (
 
 // Address provides remote address to producer
 type Address struct {
```
Comment (glaslos) [inline producer/http.go]: You also have to rename this to describe better what it does.

Comment (MohammadBilalArif) [inline producer/http.go]: May be config is batter then.

**Comment by MohammadBilalArif** [comment]
Requested changes are done :)

**Pull request merged**